ide integration broken due to invalid pom for org/sonatype/oss/oss-parent/7/oss-parent-7.pom

Before (ok):
```
Downloading: http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading: https://oss.sonatype.org/content/repositories/releases/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading: https://oss.sonatype.org/content/repositories/snapshots/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading: https://scala-ci.typesafe.com/artifactory/central/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloaded: https://scala-ci.typesafe.com/artifactory/central/org/sonatype/oss/oss-parent/7/oss-parent-7.pom (5 KB at 294.4 KB/sec)
```

Now (not ok!):
```
Downloading: https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading: https://scala-ci.typesafe.com/artifactory/scala-release-temp/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloading: http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
[WARNING] Checksum validation failed, expected This but is 504ee725835df7101a0258435141ba29954ff5a8 for http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
[WARNING] Checksum validation failed, expected This but is 504ee725835df7101a0258435141ba29954ff5a8 for http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom
Downloaded: http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom (73 B at 0.2 KB/sec)
[ERROR] The build could not read 1 project -> [Help 1]
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[FATAL] Non-parseable POM /home/jenkins/workspace/scala-2.11.x-integrate-ide/target/m2repo/org/sonatype/oss/oss-parent/7/oss-parent-7.pom: only whitespace content allowed before start tag and not T (position: START_DOCUMENT seen T... @1:1)  @ /home/jenkins/workspace/scala-2.11.x-integrate-ide/target/m2repo/org/sonatype/oss/oss-parent/7/oss-parent-7.pom, line 1, column 1
```

Because [that file](http://snapshots.repository.codehaus.org/org/sonatype/oss/oss-parent/7/oss-parent-7.pom) contains:
```
This domain doesn't resolve - so you should even be able to get this file
```

--- a/templates/default/m2-settings-public-jobs.xml.erb
+++ b/templates/default/m2-settings-public-jobs.xml.erb
@@ -48,6 +48,20 @@
       <url>     <%=node['repos']['caching-proxy']['jcenter']['url']%></url>
       <mirrorOf><%=node['repos']['caching-proxy']['jcenter']['mirrorOf']%></mirrorOf>
     </mirror>
+
+    <!--
+      codehaus doesn't exist anymore. it doesn't 404, but serves up
+      bad POMs, so we disable it globally by overriding with a mirror
+      with an invalid URL.
+      this entry could be removed after the scala-ide/scala-refactoring
+      repo removes their reference to codehaus-snapshots from their pom.xml
+    -->
+    <mirror>
+      <id>codehaus-snapshots-mirror</id>
+      <name>Maven Codehaus snapshot repository</name>
+      <url>file:///codehaus-does-not-exist-anymore</url>
+      <mirrorOf>codehaus-snapshots</mirrorOf>
+    </mirror>
   </mirrors>
 
 </settings>

--- a/templates/default/m2-settings.xml.erb
+++ b/templates/default/m2-settings.xml.erb
@@ -40,6 +40,20 @@
       <url>     <%=node['repos']['caching-proxy']['jcenter']['url']%></url>
       <mirrorOf><%=node['repos']['caching-proxy']['jcenter']['mirrorOf']%></mirrorOf>
     </mirror>
+
+    <!--
+      codehaus doesn't exist anymore. it doesn't 404, but serves up
+      bad POMs, so we disable it globally by overriding with a mirror
+      with an invalid URL.
+      this entry could be removed after the scala-ide/scala-refactoring
+      repo removes their reference to codehaus-snapshots from their pom.xml
+    -->
+    <mirror>
+      <id>codehaus-snapshots-mirror</id>
+      <name>Maven Codehaus snapshot repository</name>
+      <url>file:///codehaus-does-not-exist-anymore</url>
+      <mirrorOf>codehaus-snapshots</mirrorOf>
+    </mirror>
   </mirrors>
 
 </settings>

